package/build_boards_manager_package.sh

```diff
@@ -132,13 +123,32 @@ fi
 cat $srcdir/package/package_esp8266com_index.template.json | \
     jq "$jq_arg" > package_esp8266com_index.json
 
-# Use Github API token, if available
+# Use Github API token, when available
 curl_gh_token_arg=()
-if [ ! -z "$CI_GITHUB_API_KEY" ]; then
+if [ -z "$CI_GITHUB_API_KEY" ]; then
+    echo "curl: API key not present"
+else
+    echo "curl: API key is present"
     curl_gh_token_arg=(-H "Authorization: token $CI_GITHUB_API_KEY")
 fi
+
 # Get previous release name
-curl --silent "${curl_gh_token_arg[@]}" https://api.github.com/repos/esp8266/Arduino/releases > releases.json
+curl --silent -D curl-headers.txt "${curl_gh_token_arg[@]}" https://api.github.com/repos/esp8266/Arduino/releases > releases.json
+
+echo "======== >>> curl dumped header >>>"
+cat curl-headers.txt
+echo "======== <<< curl dumped header <<<"
+if ! grep "HTTP.*200 OK" curl-headers.txt > /dev/null; then
+    if [ -z "$CI_GITHUB_API_KEY" ]; then
+        # github api is rate-limiting
+        echo "---- Bad moon phase, in a PR, exit successfully"
+        exit 0
```

Comment:
My comment above about failing with non-zero exit code is still valid, I think. If this is started from command line, it should not pretend to succeed without doing anything useful.